In [1]:
t=0
x= int(input("sayi gir: "))
y= int(input("sayi gir: "))
t=x+y
print("toplam:",t)


sayi gir: 2
sayi gir: 4
toplam: 6


In [2]:
a=0
t= int(input("sayi gir: "))
y= int(input("sayi gir: "))
a=(t*y)/2
print("alan:",a)

sayi gir: 2
sayi gir: 3
alan: 3.0


In [4]:
sayi=5
print(sayi,"şu tiptedir: ",type(sayi))

5 sayı şu tiptedir:  <class 'int'>


In [5]:
yaslar=[17,33,26,4,58]
print(yaslar)
print("tip: ",type(yaslar))

[17, 33, 26, 4, 58]
tip:  <class 'list'>


In [6]:
yaslar=[17,33,26,4,58]
print(yaslar[0])
print(yaslar[3])

17
4


In [7]:
list=[1,"hello",3.4]
print(list[-1])
print(list[-3])

3.4
1


In [8]:
#pythonda liste dilimleme
my_list =['p','y','t','H','o','n','l','i','s','T']
#indisi 2 den 4 e kadar olanlar 
print(my_list[2:5])
#indisi 5 den sona kadar olanlar 
print(my_list[5:])
#baştan sona kadar olan tüm indisler
print(my_list[:])

['t', 'H', 'o']
['n', 'l', 'i', 's', 'T']
['p', 'y', 't', 'H', 'o', 'n', 'l', 'i', 's', 'T']


In [10]:
#demet(tuple)
#parantez içinde
urun =("microsoft","xbox",499.99)
print(urun[0])
print(urun[2])

microsoft
499.99


In [12]:
#sözlük (dictionary)
baskentler ={
    "türkiye": "ankara",
	"azerbaycan": "bakü",
    "fransa": "paris"    
}
print(baskentler["türkiye"])

ankara


In [13]:
tamsayi = 123
ondaliklisayi = 1.23
yenisayi =tamsayi +ondaliklisayi
print("sonuç:",yenisayi)
print("veri tipi: ",type(yenisayi))

sonuç: 124.23
veri tipi:  <class 'float'>


In [14]:
#örtülü tür dönüşümü
tamsayi = "123"
ondaliklisayi = 1.23
yenisayi =tamsayi +ondaliklisayi
print("sonuç:",yenisayi)
print("veri tipi: ",type(yenisayi))

TypeError: can only concatenate str (not "float") to str

In [15]:
#typcasting ya da cast denir
tamsayi = "123"
ondaliklisayi = 1.23
ondaliklitamsayi =float(tamsayi)
yenisayi =ondaliklitamsayi +ondaliklisayi
print("sonuç:",yenisayi)
print("veri tipi: ",yenisayi)

sonuç: 124.23
veri tipi:  124.23


In [16]:
#Klavyeden girilen sayıların bölüm ve kalanını bulan uygulama ?
bolen= int(input("sayi gir: "))
bolunen=int(input("sayi gir: "))
bolum=bolen//bolunen
kalan=bolen%bolunen
print("bolum: ",bolum)
print("kalan: ",kalan)

sayi gir: 10
sayi gir: 2
bolum:  5.0
kalan:  0


In [88]:
#İkinci dereceden denklem köklerini bulan bir uygulama?
print("a2+bx+c şeklindeki denklemin","köklerini bulun")
a = float(input("a'yı girin:"))
b = float(input("b'yı girin:"))
c = float(input("c'yı girin:"))

d = b**2 - 4*a*c
k1 = (-b+d**0.5)/(2*a)
k2 = (-b-d**0.5)/(2*a)
print(a,"x2",b,"x+",c,"kökleri:",k1,"ve",k2)

a2+bx+c şeklindeki denklemin köklerini bulun
a'yı girin:2
b'yı girin:4
c'yı girin:5
2.0 x2 4.0 x+ 5.0 kökleri: (-0.9999999999999999+1.224744871391589j) ve (-1-1.224744871391589j)


In [17]:
#Yıl sonu notu hesaplayan bir uygulama?
vize= int(input("not gir: "))
final=int(input("not gir: "))
yilsonu= vize *0.4 + final*0.6
print("yil sonu notu: ",yilsonu)

not gir: 50
not gir: 50
yil sonu notu:  50.0


In [23]:
#İki Değişkenimiz var bunların değerlerini nasıl takas ederiz?
ilksayi=int(input("ilk sayiyi gir: "))
ikincisayi=int(input("ikinci sayiyi gir: "))
gecicisayi=ilksayi
ilksayi=ikincisayi
ikincisayi=gecicisayi

print("ilksayi: ",ilksayi)
print("ikincisayi: ",ikincisayi)

ilk sayiyi gir: 2
ikinci sayiyi gir: 5
ilksayi:  5
ikincisayi:  2


3.DERS (Koşul ve Döngüler)

In [24]:
dogruMu = False
dogruMu =True
print(type(dogruMu))

<class 'bool'>


In [25]:
a = 3
b = 5
print(a==b) # eşit
print(a!=b) # eşit değil
print(a>b)  # büyük
print(a<b)  # küçük
print(a>=b) # büyük eşit
print(a<=b) # küçük eşit


False
True
False
True
False
True


In [26]:
print(True and False)
print(True or False)
print(not True)

False
True
False


In [27]:
sinavnotu=75
if sinavnotu>65:
    print("sınav başarılı")
else:
    print("sınav başarısız")

sınav başarılı


In [28]:
sinavnotu=75
if sinavnotu>85:
    print("aa")
elif sinavnotu >75:
    print("bb")
else:
    print("cc")

cc


In [47]:
haftaninGunu = 3
match haftaninGunu:
    case 1:
        print("pazartesi")
    case 2:
        print("salı")
    case 3:
        print("çarşamba")
    case 4:
        print("perşembe")
    case 5:
        print("cuma")
    case 6:
        print("cmt")
    case 7:
        print("pzr")
    case _:
        print("haftada 7 gün var!")

SyntaxError: invalid syntax (<ipython-input-47-ec7e9020ccb4>, line 2)

Python'da 2 tür döngü vardır:
• for döngüsü
• while döngüsü

In [33]:
dizi = ['Swift','Python','C#','Java']
#listenin her öğesi için döngü
for dil in dizi:
    print(dil)

Swift
Python
C#
Java


In [34]:
#range, iki sayısal aralık arasındaki değerler dizisidir.
for i in range(4):
	print(i)
print("---------")
for j in range(7,11):
    print(j)

0
1
2
3
---------
7
8
9
10


In [35]:
dizi = ['Swift','Python','C#','Java']
#listenin her öğesi için döngü
for i in range(4):
    print(i,".eleman:",dizi[i])

0 .eleman: Swift
1 .eleman: Python
2 .eleman: C#
3 .eleman: Java


In [36]:
# Python’da while döngüsü, belirli bir koşul karşılanana 
#kadar bir blok kodunu çalıştırmak için kullanılır

#başlangıç değerlerini atla
i = 1
n = 5
#1 den 5 e kadar while döngüsü
while i <= n:
    print(i)
    i = i+1

1
2
3
4
5


In [91]:
dizi = ['Swift','Python','C#','Java']
i = 0
n = 4
#0 dan 4 e kadar while döngüsü
while i < n:
    print(i+1,".eleman:",dizi[i])
    i = i+1

1 .eleman: Swift
2 .eleman: Python
3 .eleman: C#
4 .eleman: Java


In [11]:
#-1 girilene kadar girilen 
#sayıları toplayan program
toplam = 0
sayi = 0
print("çıkmak için -1 girabilirsiniz")
while sayi != -1:
    toplam += sayi
    sayi = int(input("sayi gir: "))
print("toplam = ", toplam)

çıkmak için -1 girabilirsiniz
sayi gir: 5
sayi gir: 7
sayi gir: 4
sayi gir: -1
toplam =  16


In [49]:
#-1 girilene kadar girilen 
#sayıları toplayan program
toplam = 0
print("çıkmak için -1 girabilirsiniz")
while True:
    sayi = int(input("sayi gir: "))
    if sayi == -1:
        break
    toplam += sayi
print("toplam = ", toplam)

çıkmak için -1 girabilirsiniz
sayi gir: 5
sayi gir: 4
sayi gir: 1
sayi gir: -1
toplam =  10


In [39]:
#Döngünün koşulu her zaman sağlanır ise döngü sonsuz kez
#çalışır (bellek dolana kadar). 

yas = 33
while yas > 18:
    print("yaş koşulu sağlandı")

yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı
yaş koşulu sağlandı


KeyboardInterrupt: 

• break deyimi, karşılaşıldığında döngüyü hemen sonlandırmak
için kullanılır.
• continue deyimi ise, döngünün geçerli yinelemesini atlamak
için kullanılır ve döngü bir sonraki yineleme başına gider.

In [45]:
for i in range(5):
    if i == 3:
        break
    print(i)


0
1
2


In [46]:
for i in range(5):
    if i == 3:
        continue
    print(i)

0
1
2
4


In [96]:
#1500 ile 2700 arasında 7'ye ve 5'in katlarına bölünebilen
#sayıları bulan Python programını yazınız.
for x in range (1500,2701):
    if(x%7==0) and (x%5==0):
        print(x)
    

1505
1540
1575
1610
1645
1680
1715
1750
1785
1820
1855
1890
1925
1960
1995
2030
2065
2100
2135
2170
2205
2240
2275
2310
2345
2380
2415
2450
2485
2520
2555
2590
2625
2660
2695


In [52]:
""" Santigrat ve Fahrenhayt cinsinden sıcaklıkları dönüştürmek
için bir Python programı yazın.
• c/5 = f-32/9 (c , santigrat cinsinden sıcaklık; f, fahrenhayt
cinsinden sıcaklık)
"""
sicaklik = input("sicaklık girin:")
derece =int(sicaklik[:-1])
tip=sicaklik[-1]
match tip:
    case "c":
        result=(9 * derece)/5 +32
        print(derece,"c= ",result,"f dir")
    case "f":
        result=(derece - 32)*5 /9
        print(derece,"c= ",result,"f dir")
    case _:
        print("uyumsuz giriş")


SyntaxError: invalid syntax (<ipython-input-52-967e3ca5313b>, line 9)

In [53]:
"""100’den 400’e kadar olan sayılardan tüm rakamları çift olan
sayıları ekrana yazdıran program."""
for i in range(100,401):
    s = str(i)
    if (int(s[0])%2==0) and (int(s[1])%2==0) and (int(s[2])%2==0):
        print(s)

200
202
204
206
208
220
222
224
226
228
240
242
244
246
248
260
262
264
266
268
280
282
284
286
288
400


In [55]:
"""Bir köpeğin yaşını köpek yılı cinsinden hesaplayan bir Python
programı yazın.
• Bilgi: İlk iki yıl için bir köpek yılı 10,5 insan yılına eşittir.
Bundan sonra her köpek yılı 4 insan yılına eşittir.
"""
iYasi = int(input("köpek yaşı gir:"))
if iYasi<0:
    print("yaş pozitif olmalı")
elif iYasi <= 2:
    print("insan yaşı karşılığı: ",iYasi *10.5)
else:
     print("insan yaşı karşılığı: ",21+(iYasi-2)*4)

köpek yaşı gir:5
insan yaşı karşılığı:  33


In [57]:
"""• Girilen bir sözcükteki sesli ve sessiz harflerin sayısını bulan bir
program yazın."""

sesli=0
sessiz=0
sozcuk =input("sözcük gir: ")
for harf in sozcuk:
    if (harf=='a' or harf=='e' or harf=='ı' or harf=='i' or harf=='o' or harf=='ö' or harf=='u' or harf=='ü'):
        sesli +=1
    else:
        sessiz+=1
print("sesli: ",sesli)
print("sessiz: ",sessiz)

sözcğk gir: nazlı
sesli:  2
sessiz:  3


In [27]:
sesli=0
sessiz=0
sozcuk =input("sözcük gir: ")
for harf in sozcuk:
    if harf in ('a','e','ı','o','ö','i','u','ü'):
        sesli +=1
    else:
        sessiz+=1
print("sesli: ",sesli)
print("sessiz: ",sessiz)

sözcük gir: hvsknlşdöclvspeılsdmzaşaaaaa
sesli:  9
sessiz:  19


In [59]:
"""Kenarları girilen üçgenin eşkenar, ikizkenar ya da çeşitkenar
olduğunu ekrana yazan program yazın."""
print("üçgen kenar uzunluklarını girin: ")
x=int(input("x: "))
y=int(input("y: "))
z=int(input("z: "))
if x==y and y==z:
    print("eşkenar üçgen")
elif x==y or y==z or x==z: 
    print("ikizkenar üçgen")
else:
    print("çeşitkenar üçgen")

üçgen kenar uzunluklarını girin: 
x: 5
y: 5
z: 5
eşkenar üçgen


In [60]:
"""• Girilen 3 sayıdan büyüklüklerine göre ortada olan sayıyı
bulan program yazın."""
a=int(input("ilk sayi: "))
b=int(input("ikinci sayi: "))
c=int(input("üçüncü sayi: "))
if a>b:
    if a<c:
        orta=a
    elif b>c:
        orta=b 
    else:
        orta=c
else:
    if a>c:
        orta=a
    elif b<c:
        orta=b 
    else:
        orta=c
print("orta sayı: ",orta)

ilk sayi: 5
ikinci sayi: 1
üçüncü sayi: 4
orta sayı:  4


In [ ]:
sayi1 = int(input("Birinci sayıyı girin: "))
sayi2 = int(input("İkinci sayıyı girin: "))
sayi3 = int(input("Üçüncü sayıyı girin: "))

if sayi1 < sayi2 < sayi3 or sayi3 < sayi2 < sayi1:
    ortadaki_sayi = sayi2
elif sayi2 < sayi1 < sayi3 or sayi3 < sayi1 < sayi2:
    ortadaki_sayi = sayi1
else:
    ortadaki_sayi = sayi3
print("Ortadaki sayı:", ortadaki_sayi)

In [61]:
#Girilen rakam için çarpım tablosu oluşturan program yazın
n=int(input("rakam gir: "))
for i in range(1,11):
    print(n,"x",i,"=",n*i)

rakam gir: 5
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


In [115]:
sayi=int(input("rakam gir: "))
toplam=0
for i in range(1,sayi):
    if (sayi%i==0):
         toplam +=i
if toplam==sayi:
    print("mükemmel syi")
else:
    print("mükemmel syi değil")

rakam gir: 14
mükemmel syi değil


In [62]:
"""Pozitif bölenlerinin toplamı kendisine eşit olan sayılara
mükemmel sayı denir.
• Girilen bir sayının mükemmel sayı olup olmadığını kontrol
eden program yazın.
• 6’nın bölenleri: 1, 2, 3
• 28’in bölenleri: 1, 2, 4, 7, 14 """
sayi =int(input("sayi gir:"))
toplam=0
for i in range (1,sayi):
    if(sayi%i==0):
        toplam+=i
if(sayi == toplam):
    print("mükemmel sayi")
else:
    print("mükemmel sayi değil")

sayi gir:14
mükemmel sayi değil


In [64]:
"""• Girilen sayının faktöriyelini hesaplayan programı yazın."""
sayi =int(input("sayi gir:"))
faktoriyel = 1
i=1
if sayi >= 0:
    while i <=sayi:
        faktoriyel *=i
        i+=1
    print(sayi,"!=",faktoriyel)
else:
    print("negatif sayıların faktöriyeli olmaz")

sayi gir:5
5 != 120


In [120]:
sayi =int(input("sayi gir:"))
faktoriyel = 1
for i in range (1,sayi+1):
    faktoriyel *= i   
print(faktoriyel)

sayi gir:5
120


In [65]:
"""• Girilen sayıyı tersten yazan program yazın."""
n =int(input("sayi gir:"))
ters=0
while(n!=0):
    sil=n%10
    ters=ters*10+sil
    n=n//10
print(ters)

sayi gir:104
401


In [134]:
sayi = input("Bir sayı girin: ")
sayiStr = str(sayi)
sayac = 0
for s in sayiStr:
    sayac += 1
negatifsayac = -1
for i in range(0,sayac ):
    print(sayiStr[negatifsayac],end="")
    negatifsayac-= 1

Bir sayı girin: 6848541
1458486

In [121]:
sayi = input("Bir sayı girin: ")
ters_sayi = sayi[::-1]
print("Girilen sayı:", sayi)
print("Tersten yazılmış sayı:", ters_sayi)

Bir sayı girin: 104
Girilen sayı: 104
Tersten yazılmış sayı: 401


In [33]:
sayi = input("Bir sayı girin: ")
ters_sayi = sayi[::-1]
print("Girilen sayının tersi:", ters_sayi)

Bir sayı girin: 401
Girilen sayının tersi: 104


4.DERS (Fonksiyon ve Modüller)

In [66]:
"""return (isteğe bağlı) - bir fonksiyondan değer
döndürür"""
def sayilariTopla(sayi1,sayi2):
    toplam=sayi1 +sayi2
    print("toplam:",toplam)
sayilariTopla(5,7)

toplam: 12


In [67]:
def sayilariTopla(sayi1,sayi2):
    toplam=sayi1 +sayi2
    print("toplam:",toplam)
sayilariTopla(sayi2=5,sayi1=7)

toplam: 12


In [68]:
def sayilariTopla(sayi1=5,sayi2=7):
    toplam=sayi1 +sayi2
    print("toplam:",toplam)
sayilariTopla(3,2)
sayilariTopla(sayi2=15)
sayilariTopla()

toplam: 5
toplam: 20
toplam: 12


In [69]:
def sayilariTopla(*sayilar):
    sonuc=0
    for sayi in sayilar:
        sonuc = sonuc + sayi
    print("toplam:",sonuc)
sayilariTopla(1,2,3)
sayilariTopla(4,9)
sayilariTopla(7)
sayilariTopla()

toplam: 6
toplam: 13
toplam: 7
toplam: 0


In [71]:
"""Eğer fonksiyonumuzun bir fonksiyon çağrısına bir
değer döndürmesini istiyorsak, "return" deyimini
kullanırız.

"return" deyimi aynı zamanda işlevin sona
erdiğini de belirtir. Dönüşten sonra herhangi bir
kod yürütülmez."""

def sayilariTopla(sayi1,sayi2):
    toplam=sayi1 +sayi2
    return toplam
print(sayilariTopla(3,5))

8


In [72]:
def KareHesapla(sayi):
    sonuc=sayi**2
    return sonuc
kare=KareHesapla(9)
print(kare)

81


•sqrt() - bir sayının karekökünü döndürür
•pow() - bir sayının kuvvetini döndürür
• Bunları kullanabilmek için modülü programımıza dahil
etmemiz gerekir.Örneğin sqrt() math modülünün içinde
tanımlanmıştır.

In [34]:
import math 
karekok = math.sqrt(25)
print("25'in karekökü: ",karekok)
kuvvet = math.pow(5,7)
print("5'in 7. kuvveti: ",kuvvet)

25'in karekökü:  5.0
5'in 7. kuvveti:  78125.0


In [36]:
#kendisini çağıran fonksiyon özyinelemelidir.maliyetli
#özyineleme ile faktöriyel çözümü
def faktoriyel(x):
    if x == 1:
        return 1
    else:
        return (x*faktoriyel(x-1))
sayi = 5
print(sayi, "!=", faktoriyel(sayi))
        

5 != 120


In [37]:
#özyineleme olmadan
"""• Girilen sayının faktöriyelini hesaplayan programı yazın."""
sayi =int(input("sayi gir:"))
faktoriyel = 1
i=1
if sayi >= 0:
    while i <=sayi:
        faktoriyel *=i
        i+=1
    print(sayi,"!=",faktoriyel)
else:
    print("negatif sayıların faktöriyeli olmaz")

sayi gir:5
5 != 120


In [38]:
a=7
def arttir():
    a+=1
    print(a)
arttir()

UnboundLocalError: local variable 'a' referenced before assignment

In [39]:
a = 7
def arttir():
    global a
    a+=1
    print(a)
arttir()

8


In [40]:
def main():
    print("merhaba")
if __name__=="__main__":
    main()

merhaba


In [41]:
#modulum.py olarak kaydet ve 
def topla(a,b):
    sonuc = a + b
    return sonuc
#sonrasında yeni dosyada 
import modulum 
print(modulum.topla(7,6))

In [42]:
import math 
print("pi değeri: ", math.pi)

pi değeri:  3.141592653589793


In [43]:
#yeniden adlandırılabilir
import math as matematik
print(matematik.pi)

3.141592653589793


In [45]:
"""
Parametre olarak sayı alan ve sayının asal olup
olmadığını kontrol eden bir Python fonksiyonu
yazınız.
"""
def asalMi(n):
    if(n == 1):
        return False
    elif(n == 2):
        return True
    else:
        for x in range(2,n):
            if(n % x == 0):
                return False
        return True
print(asalMi(23))

True


In [7]:
def asalMi(sayi):
    for i in range(2,sayi):
        if(sayi%i==0):
            print("sayi asal değil")
    print("sayı asal")
asalMi(25)

sayi asal değil
sayı asal


In [9]:
def asalMi(sayi):
    for i in range(2,sayi):
        if(sayi%i==0):
            print("sayi asal değil")
            return
    print("sayı asal")
asalMi(25)

sayi asal değil


In [12]:
def asalMi(sayi):
    asal = True
    for i in range(2,sayi):
         if(sayi % i == 0):
            asal = False
            break
    if(asal):
        print("sayi asal")
    else:
        print("sayi asal değil")
asalMi(23)

sayi asal


In [158]:
"""Girilen bir dizenin palindrom olup olmadığını
kontrol eden bir Python fonksiyonu yazın."""
def polindrom(dize):
    dize = dize.lower()
    return dize==dize[::-1]
metin = input("metin girin: ")    
if palindrom(metin):
    print("palindrom")
else:
    print("değil")

metin girin: madam
palindrom


In [49]:
"""Girilen bir sayının Palindrome olup olmadığını
bulan program yazın."""
def palindromeMu(sayi):
    gecici = sayi
    ters = 0
    while (sayi != 0):
        k= sayi%10
        ters=ters*10+k
        sayi = sayi//10
    if (ters ==gecici):
        print(gecici, "palindrome sayıdır")
    else:
        print(gecici, "palindrome değil")
palindromeMu(int(input("sayi gir: ")))

sayi gir: 505
505 palindrome sayıdır


In [167]:
def palindrom_mu(sayi1):
    str_sayi = str(sayi1)
    return str_sayi == str_sayi[::-1]
sayi = int(input("Bir sayı girin: "))
if palindrom_mu(sayi):
    print(sayi, "palindrome sayıdır")
else:
    print(sayi, "palindrome değil")


Bir sayı girin: 545
545 palindrome sayıdır


In [13]:
"""2 sayı değeri alan ve bunların toplamını, farkını ve
çarpımını döndüren bir fonksiyon tanımlayın"""
def hesapla(a,b):
    toplam = a+b
    fark = a-b
    carpim = a*b
    print("toplam: ",toplam)
    print("fark: ",fark)
    print("carpim: ",carpim)
a = int(input("birinci sayi: "))
b = int(input("ikinci sayi: "))
hesapla(a,b)

birinci sayi: 5
ikinci sayi: 4
toplam:  9
fark:  1
carpim:  20


In [15]:
def fonksiyonum(sayi1,sayi2):
    print("toplam: ",sayi1+sayi2)
    print("fark: ",sayi1-sayi2)
    print("çarpım: ",sayi1*sayi2)
fonksiyonum(3,7)

toplam:  10
fark:  -4
çarpım:  21


In [59]:
"""Özyinelemeyi kullanarak bir sayının kuvvetini bulan
Python programını yazın."""
def kuvvet(taban, us):
    if us == 0:
        return 1
    else:
        return taban * kuvvet(taban, us - 1)
taban = int(input("Tabanı girin: "))
us = int(input("Üssü girin: "))
print(kuvvet(taban,us))


Tabanı girin: 5
Üssü girin: 1
5


In [17]:
"""Özyinelemeyi kullanarak bir sayının kuvvetini bulan
Python programını yazın."""
def kuvvetBul(sayi,kuvvet):
    if kuvvet==0:
        return 1
    sonuc =sayi*kuvvetBul(sayi,kuvvet-1)
    return sonuc 

sayi=int(input("sayı: "))
kuvvet=int(input("kuvvet: "))
kuvvetBul(sayi,kuvvet)

sayı: 5
kuvvet: 3


125

In [62]:
def kuvvet(taban, us):
    if us == 0:
        return 1
    return (taban*kuvvet(taban,us-1))
if __name__== '__main__':
    taban=int(input("taban girin: "))
    us=int(input("us girin: "))
    print(kuvvet(taban,us))

taban girin: 5
us girin: 4
625


In [1]:
"""Girilen sayılardan en büyük ve en küçüğünü ekrana
yazdıran fonksiyonu yazın. 
•Fonksiyona istenildiği kadar sayı girilebilecektir"""
def minMaxYaz(*sayilar):
    min = 99999999999999
    max = 0
    for sayi in sayilar:
        if sayi>max:
            max=sayi
        if sayi<min:
            min=sayi
    print("min: ",min)
    print("max: ",max)
minMaxYaz(3,5,8,5,7,36,79,-1)

min:  -1
max:  79


In [4]:
"""Belirli bir listenin veya dizenin uzunluğunu
hesaplayan bir fonksiyon tanımlayın. """
def uzunluk(dizi):
    sayac=0
    for x in dizi:
        sayac += 1
    return sayac
metin = "merhaba arkadaşlar python dersine hoşgeldiniz"
print(uzunluk(metin))
liste=[1,2,3,4,5,6,7]
print(uzunluk(liste))

45
7


In [18]:
def uzunlukBul(liste):
    sayac=0
    for eleman in liste:
        sayac += 1
    return sayac

def enUzunuBul(liste):
    enUzunMetin = liste[0]
    for metin in liste:
        if uzunlukBul(metin)>uzunlukBul(enUzunMetin):
            enUzunMetin=metin
    print(enUzunMetin)
metinler = ["python","ahmet","arel","swift","dlnbdlsndlsnml"]
enUzunuBul(metinler)

dlnbdlsndlsnml


In [5]:
"""Metinlerden oluşan bir dizide en uzun metni bulan
program yazın."""
def enUzunMetin(metinler):
    max=0
    metin=""
    for y in metinler:
        if max<uzunluk(y):
            max=uzunluk(y)
            metin=y
    print(metin," ",max, " karakterle en uzun")
metinListesi = ["merhaba","ankara","python dersi"]
enUzunMetin(metinListesi)

python dersi   12  karakterle en uzun


In [ ]:
#alternatif
def en_uzun_metni_bul(metinler):
    en_uzun_metin = ""
    for metin in metinler:
        if len(metin) > len(en_uzun_metin):
            en_uzun_metin = metin
    return en_uzun_metin
metin_dizisi = ["Python", "Java", "C++", "JavaScript", "Ruby"]
uzun_metin = en_uzun_metni_bul(metin_dizisi)
print("En uzun metin:", uzun_metin)

In [19]:
"""
•Sezar şifrelemesi, ilk kez Romalı lider Jül Sezar
tarafından kullanılmış olan bir şifreleme tekniğidir.
•Bu şifrede, her harf o harften birkaç sonraki harf
kullanılarak yazılır.
•Örneğin, 3 harf atlamalı Sezar Şifresi'nde "deneme"
yerine "ijtjşj" yazılır.
"""
harfler= ["a","b","c","ç","d","e","f","g","ğ","h","ı","i","j","k","l","m","n","o","ö","p","r","s","ş","t","u","ü","v","y","z"]
def sifrele(metin,atlamaSayisi):
    for k in metin:
        sayac=0
        for h in harfler:
            if k ==h:
                break
            sayac +=1
        yeniIndis=(sayac+atlamaSayisi-1)%29
        print(harfler[yeniIndis],end="")
def coz(metin,atlamaSayisi):
    for k in metin:
        sayac=0
        for h in harfler:
            if k==h:
                break
            sayac+=1
        yeniIndis=(sayac+atlamaSayisi-1)%29
        print(harfler[yeniIndis],end="")
        
sifrele("ahmetözcan",8)
print("")
coz("gnşjbüfhgt",8)

gnşjbüfhgt
ltapğçknlb

5. ders

In [20]:
sayilar=[21,34,54,12]
print("ekleme öncesi: ",sayilar)
sayilar.append(32)
print("ekleme sonrası: ",sayilar)

ekleme öncesi:  [21, 34, 54, 12]
ekleme sonrası:  [21, 34, 54, 12, 32]


In [21]:
sayilar=[1,3,5]
ciftSayilar=[2,4,6]
sayilar.extend(ciftSayilar)
print("eklenmiş liste: ",sayilar)

eklenmiş liste:  [1, 3, 5, 2, 4, 6]


In [32]:
sayilar=[1,3,5]
ciftSayilar=[2,4,6]
sayilar.append(ciftSayilar)
print("eklenmiş liste: ",sayilar)

eklenmiş liste:  [1, 3, 5, [2, 4, 6]]


In [22]:
sayilar=[10,30,40]
sayilar.insert(1,20)
print(sayilar)

[10, 20, 30, 40]


In [23]:
diller=["python","swift","c"]
diller[2]="c++"
print(diller)

['python', 'swift', 'c++']


In [24]:
diller=["python","swift","c"]
del diller[1]
print(diller)

['python', 'c']


In [25]:
diller=["python","swift","java","c"]
diller.remove("python")
print(diller)

['swift', 'java', 'c']


öğe ekleme veya öğe kaldırma yöntemleri Tuple'da
mevcut değildir
• Demetler değişmez olduğundan, bir demet boyunca yineleme
yapmak listeye göre daha hızlıdır. Yani hafif bir performans artışı sağlar.

In [26]:
demet=("a","p","p","l","e")
print(demet.count("p"))
print(demet.index("l"))
print(len(demet))

2
3
5


dizeler değişmezdir. 

In [27]:
ileti="merhaba"
print(ileti[0])
ileti[0]="h"
print(ileti)

m


TypeError: 'str' object does not support item assignment

In [34]:
ileti="merhaba"
ad="h" +ileti[1:]
print(ad)

herhaba


In [28]:
ornek="bu bir alıntıdır, "O da ne?""
print(ornek)

SyntaxError: invalid syntax (<ipython-input-28-69b461146bbe>, line 1)

In [30]:
ornek="bu bir alıntıdır, \"O da ne?\""
print(ornek)

bu bir alıntıdır, "O da ne?"


In [31]:
ders="python"
konu="stringler"
print(f"{ders} dersinde konumuz: {konu}")

python dersinde konumuz: stringler


Bir listenin ilk ve son elemanını değiştirecek bir fonksiyon yazıp sınayın.
• Girdi : [12, 35, 9, 56, 24]
• Çıktı : [24, 35, 9, 56, 12]
• Girdi : [1, 2, 3]
• Çıktı : [3, 2, 1]


In [35]:
def degis(yeniListe):
    size = len(yeniListe)
    gecici = yeniListe[0]
    yeniListe[0]=yeniListe[size-1]
    yeniListe[size-1]=gecici
    return yeniListe
liste=[12,35,9,56,24]
print(degis(liste))

[24, 35, 9, 56, 12]


In [36]:
#dersteki
def degis(liste):
    gecici=liste[0]
    liste[0]=liste[-1]
    liste[-1]=gecici
    return liste 
print(degis([12,35,9,56,24]))

[24, 35, 9, 56, 12]


In [38]:
def palindromMu(sozcuk):
    sozcuk=list(sozcuk)
    tersSozcuk=sozcuk.copy()
    tersSozcuk.reverse()
    if tersSozcuk==sozcuk:
        return True
    else:
        return False
print(palindromMu("aga"))

True


In [41]:
"""Verilen listenin öğelerinin toplamını ve aritmetik ortalamasını bulan
fonksiyonu yazıp sınayın.
• Liste sadece sayılardan oluşacaktır.
"""
def toplamVeOrtalama(liste):
    toplam=0
    for i in liste:
        toplam += i
    ortalama=toplam/len(liste)
    print("toplam: ",toplam)
    print("ortalama: ",ortalama)
L=[4,5,1,2,9,7,10,8]
toplamVeOrtalama(L)

toplam:  46
ortalama:  5.75


In [43]:
import numpy as np
def hesapla(liste):
    print("toplam: ",np.sum(liste))
    print("ortalama: ",np.average(liste))
hesapla([3,5,7,1,3])

toplam:  19
ortalama:  3.8


In [42]:
import numpy as np
def toplamVeOrtalama(liste):
    toplam=np.sum(liste)
    ortalama=np.average(liste)
    print("toplam: ",toplam)
    print("ortalama: ",ortalama)
L=[4,5,1,2,9,7,10,8]
toplamVeOrtalama(L)

toplam:  46
ortalama:  5.75


Bir listedeki en küçük ve en büyük sayıyı bulan fonksiyonu yazıp sınayın.
• Liste sadece sayılardan oluşacaktır.

In [44]:
def minMaxBul(liste):
    liste.sort()
    print("min: ",liste[0])
    print("max: ",liste[-1])
L=[4,5,1,2,9,7,10,8]
minMaxBul(L)

min:  1
max:  10


In [45]:
import numpy
def enKucukEnBuyuk(liste):
    print("min: ",numpy.min(liste))
    print("max: ",numpy.max(liste))
enKucukEnBuyuk([3,5,7,1,3])

min:  1
max:  7


Bir listedeki benzersiz (uniqe) öğelerin sayısını bulan fonksiyonu yazıp
sınayın.
• Liste sadece sayılardan oluşacaktır.

In [48]:
def benzersiz(liste):
    geciciliste=[]
    sayac=0
    for item in liste:
        if item not in geciciliste:
            sayac +=1
            geciciliste.append(item)
    return sayac
L=[1,2,2,5,8,44,44,8]
print(benzersiz(L))

5


• Bir demetteki öğelerin kaç kere tekrar ettiğini bulan bir fonksiyon yazıp
sınayın

In [186]:
def frekans(demet):
    tekrar = {}
    for eleman in demet:
        if eleman in tekrar:
            tekrar[eleman] += 1
        else:
            tekrar[eleman] = 1
    return tekrar
ornek = (1, 2, 3, 1, 2, 1, 4, 5, 4, 3, 6)
frekans(ornek)

{1: 3, 2: 2, 3: 2, 4: 2, 5: 1, 6: 1}

In [56]:
#dersteki
def frekans(demet):
    frek_dict={}
    for sayi in demet:
        if sayi in frek_dict.keys():
            frek_dict[sayi] +=1
        else:
            frek_dict[sayi]=1
    print(frek_dict)
frekans([3,5,4,3,3,3,4,1])

{3: 4, 5: 1, 4: 2, 1: 1}


• Bir metindeki harflerin kaç kere tekrar ettiğini bulan bir fonksiyon yazıp
sınayın.

In [65]:
def harfFrekansi(metin):
    frek={}
    for harf in metin:
        if harf in frek:
            frek[harf] += 1
        else:
            frek[harf] = 1
    print(frek)
harfFrekansi("nesne yönelimli python programlama")

{'n': 4, 'e': 3, 's': 1, ' ': 3, 'y': 2, 'ö': 1, 'l': 3, 'i': 2, 'm': 3, 'p': 2, 't': 1, 'h': 1, 'o': 2, 'r': 2, 'g': 1, 'a': 3}


Geçerli şifre girişi yapılmadan işlemlere başlamayan bir program yazın.
• Program açılışta şifre soracak, geçerli şifre girilirse devam edecek, 3 kere
hatalı şifre girilmesi durumunda programdan çıkılacaktır.


In [66]:
def programIslemleri():
    print("programa hoşgeldin")
sifresayac=0
while(True):
    sifre=input("lütfen şifre girin: ")
    if sifre.lower()=="python":
        programIslemleri()
        break
    else:
        sifresayac += 1
        if sifresayac==3:
            print("şifreyi 3 kere hatalı girdiniz")
            break

lütfen şifre girin: sdmhfcvsk
lütfen şifre girin: smnbdcms
lütfen şifre girin: ncdsö
şifreyi 3 kere hatalı girdiniz


Girilen metinde her sözcüğün baş harfini büyük diğer karakterlerini küçük
hale getiren programı yazınız.

In [69]:
def buyuk(metin):
    sozcukler=metin.split(" ")
    buyukharf=""
    for sozcuk in sozcukler:
        gecici=sozcuk[0].upper()+sozcuk[1:].lower()
        buyukharf=buyukharf+" "+gecici
    return buyukharf
print(buyuk("deENEME ifadEESİ BUraya GeleCEK"))

 Deeneme Ifadeesi̇ Buraya Gelecek


In [70]:
def ilkHrfBuyuk(metin):
    return metin.title()
print(ilkHrfBuyuk("haTALI şiFRE HAK KlamaDI"))

Hatali Şifre Hak Klamadi


• Verilen bir ifadeyi var olan metinin ortasına ekleyen bir program yazın.

In [198]:
def ekle (metin2,ifade):
    return metin2[:len(metin2)//2]+ifade +metin2[len(metin2)//2:]
print(ekle("python","pro"))

pytprohon


Girilen ondalıklı sayıyı 2 basamak hassasiyeti ile yuvarlayan bir program yazın.
• 3.1458755877 => 3.14
• 12.75987568755 => 17.76

In [73]:
def yuvarla(sayi):
    ayrik = sayi.split(".")
    tam = ayrik[0]
    ondalik = ayrik[1]
    yeniOndalik=""
    if int(ondalik[1])==9:
        yeniOndalik= str(int(ondalik[0])+1)+"0"
    elif int(ondalik[1])==9:
        yeniOndalik= ondalik[0] + str(int(ondalik[1])+1)
    else:
        yeniOndalik = ondalik[0]+ondalik[1]
    return tam+"."+yeniOndalik
print(yuvarla("3.1458755877"))

3.14


In [74]:
def yuvarla (sayi):
    return round(float(sayi),2)
print(yuvarla("3.1458755877"))

3.15


Girilen metindeki sesli harfleri silen bir program yazın.

In [75]:
def sessizHarfCikar(metin):
    yeniMetin=""
    for harf in metin:
        if harf not in ("a","e","ı","i","o","ö","u","ü"):
            yeniMetin +=harf
    return yeniMetin
print(sessizHarfCikar("merhaba nasılsın naber"))

mrhb nslsn nbr


 Sayıların metinsel ifadesini içeren metinden sayı üreten program yazın.
• dört üç iki beş => 4325
• yedi sekiz altı dokuz =>7869

In [86]:
def sayiyaCevir(metini):
    rakamlar={"sıfır":"0","bir":"1","iki":"2","üç":"3","dört":"4","beş":"5","altı":"6","yedi":"7","sekiz":"8","dokuz":"9"}
    sayiMetinleri=metini.split()
    yeniSozcuk = ""
    for i in sayiMetinleri:
        yeniSozcuk += rakamlar[i]
    return yeniSozcuk
print(sayiyaCevir("dört üç beş iki"))

4352


6.ders

Girilen Sayının Tek mi Çift mi Olduğunu Bulma

In [204]:
def tekMi(sayi):
    
    if sayi%2==0:
        print("çift sayi")
    else:
        print("tek sayi")
tekMi(int(input("sayi gir: ")))

tek sayi


Girilen Sayının Pozitif Olup Olmadığını Bulma

In [212]:
def pozitifMi(sayi):
    if sayi<0:
         print("sayi negatiftir")
    elif sayi>0:
         print("sayi pozitiftir")
    else:
         print("sayı işaretsiz")
pozitifMi(int(input("sayi gir: ")))         

sayi gir: 0
sayı işaretsiz


In [ ]:
Girilen Sayının Mutlak Değerini Bulma

In [222]:
def mutlak(sayi):
    if sayi<0:
         print("|",sayi,"|=",sayi*-1)
    else: 
         print(sayi)
mutlak(int(input("sayi gir: "))) 

sayi gir: -8
| -8 |= 8


In [221]:
sayi=int(input("sayi gir: "))
mutlak=sayi
if sayi<0:
    mutlak=sayi*-1
print("|",sayi,"|=",mutlak)

sayi gir: -5
| -5 |= 5


In [ ]:
EBOB ve EKOK Bulma

ebob(sayi1,sayi2) x ekok(sayi1,sayi2) = sayi1 * sayi2

In [223]:
def ebob_ekok(birincisayi,ikincisayi):
    if birincisayi > ikincisayi:
        kucuksayi = ikincisayi
    else:
        kucuksayi = birincisayi
    for i in range (1,kucuksayi+1):
        if (birincisayi % i == 0) and (ikincisayi%i==0):
            ebob=i
    ekok = (birincisayi*ikincisayi) // ebob
    return ebob,ekok
birincisayi = int(input("birincisayi gir: "))
ikincisayi = int(input("ikincisayi gir: "))
ebob,ekok = ebob_ekok(birincisayi,ikincisayi)
print("ebob: ",ebob)
print("ekok: ",ekok)

birincisayi gir: 6
ikincisayi gir: 7
ebob:  1
ekok:  42


In [ ]:
Vücut Kitle İndeksi Hesaplama

vki = kilo(kg)/boy(cm) 2
vki< 18,5 ise Zayıf
18,5 < vki < 24,9 ise Normal
25 < vki < 29,9 ise Fazla kilolu
30 < vki < 34,9 ise 1. derece obez
35 < vki < 39,9 ise 2. derece obez
40 < vki ise 3. derece obez

In [245]:
def vki(boy, kilo):
    vki = kilo / (boy ** 2)

    if vki < 18.5:
        print("Zayıf")
    elif vki < 25:
        print("Normal")
    elif vki < 30:
        print("Fazla kilolu")
    elif vki < 35:
        print("1. derece obez")
    elif vki < 40:
        print("2. derece obez")
    else:
        print("3. derece obez")
boy = float(input("Boy gir: "))
kilo = float(input("Kilo gir: "))

vki(boy, kilo)


Boy gir: 1.73
Kilo gir: 60
Normal


In [ ]:
Armstrong Sayısı Bulma

Bir sayının, tüm basamaklarındaki rakamların sayının basamak katına toplamı kendisine eşit olan sayılara Armstrong sayısı denir.
Örneğin; 407 sayısını ele alalım. 407’yi rakamlarına ayıralım. Sonrasında da bu rakamları, sayı üç haneli olduğu için rakamların küplerini alıp toplayalım:
407 => (4)3 + (0)3 + (7)3 = 407 yapmaktadır. Yani bu demektir ki 407 sayısı Armstrong bir sayıdır. 


In [246]:
def armstrongMu(sayi):
    uzunluk=len(sayi)
    toplam=0
    for rakam in sayi:
        toplam += int(rakam)**uzunluk
    if toplam==int(sayi):
        print(sayi,"armstrong")
    else: 
        print(sayi,"armstrong değil")
armstrongMu(input("sayi gir: "))

sayi gir: 407
407 armstrong


In [ ]:
FizzBuzz oyunu

1’den 100’e kadar sayılar yazdırılacak
Sayı eğer 3’e bölünebiliyorsa, (örneğin; 3) sayıyı yerine “Fizz” yazacak
Sayı eğer 5’e bölünebiliyorsa, (örneğin; 5) sayıyı yerine “Buzz” yazacak
Sayı eğer hem 3’e hem de 5’e bölünebiliyorsa, (örneğin; 15) sayı yerine “FizzBuzz” yazacak

In [249]:
for i in range(1,20):
    if i%3==0 and i%5==0:
        print("fizzBuzz")
        continue
    if i%3==0:
        print("fizz")
        continue
    if i%5==0:
        print("buzz")
        continue
    print(i)

1
2
fizz
4
buzz
fizz
7
8
fizz
buzz
11
fizz
13
14
fizzBuzz
16
17
fizz
19


In [ ]:
Basit hesap makinesi

Sadece iki sayı alacak ve dört işlemden (+,-,*,/) birini gerçekleştirebilecek
3+2
4-1
9/3
7*5

In [250]:
def hesapYap():
    girdi = input("Lütfen işleminizi girin:")
    islemler = ["+","-","*","/"]
    for islem in islemler:
        girdiler = girdi.split(islem)
        if len(girdiler)>=2:
            sayi1 = girdiler[0]
            sayi2 = girdiler[1]
            match islem:
                case "+":
                    print(sayi1,islem,sayi2,"=",int(sayi1)+int(sayi2))
                case "-":
                    print(sayi1,islem,sayi2,"=",int(sayi1)-int(sayi2))            
                case "*":
                    print(sayi1,islem,sayi2,"=",int(sayi1)*int(sayi2))
                case "/":
                    print(sayi1,islem,sayi2,"=",int(sayi1)/int(sayi2))
            break


SyntaxError: invalid syntax (<ipython-input-250-25a15f9fa3ab>, line 9)

In [ ]:
Sayı Tahmini Oyunu
1 ile 100 arasında rastgele bir sayı belirle 
random modülünü kullan
random.randint(sayi1, sayi2) #iki sayı arasında rastgele bir sayı belirler
Kullanıcının tahmini bilgisayarın tuttuğu sayıdan büyükse, küçük bir sayı tutmasını söyle, değilse tam tersini yap
Kaç denemede kullanıcının sayıyı bulduğunu yaz


In [254]:
import random
tutulansayi=random.randint(1,100)
tahminedilensayi=-1
denemesayisi=0
while tutulansayi !=tahminedilensayi:
    tahminedilensayi=int(input("sayi gir: "))
    denemesayisi +=1
    if tahminedilensayi>tutulansayi:
        print("bilemedin daha küçük bir sayı gir")
    elif tahminedilensayi<tutulansayi:
        print("bilemedin daha büyük bir sayi gir")
    else:
        print("tebrikler",denemesayisi," seferde bildin")

KeyboardInterrupt: Interrupted by user

In [ ]:
Yapılacaklar Listesi
Kullanıcının yapması gereken görevleri ekleyebileceği, görebileceği ve silebileceği bir yapılacaklar listesi uygulaması

In [267]:
yapilacaklar=[]
def gorevekle(gorev):
    yapilacaklar.append(gorev)
    print(gorev," eklendi")
def gorevlerigor():
    print(yapilacaklar)
def gorevsil(gorev):
    yapilacaklar.remove(gorev)
    print(gorev," silindi")

while True:
    print("\n\nseçim yap: ")
    print("1- görev ekle: ")
    print("2- görevleri gör: ")
    print("3- görev sil: ")
    print("4- çık\n")
    giris=input("seçiminiz: ")
    if giris == "1":
        gorevekle(input("eklemek istediğin görevler: "))
    elif giris == "2":
        gorevlerigor()
    elif giris=="3":
        gorevsil(input("silmek istediğin görevler: "))
    elif giris=="4":
        print("çıkılıyor....")
        break
    else:
        print("hatalı seçim")



seçim yap: 
1- görev ekle: 
2- görevleri gör: 
3- görev sil: 
4- çık



KeyboardInterrupt: Interrupted by user

Taş Kağıt Makas Oyunu
Kullanıcıdan seçimini alıp bilgisayarın yaptığı rastgele seçimle karşılaştır ve sonucu yaz
Rastgele sayı üretmeyi hatırla
Taş > Makas
Makas > Kağıt
Kağıt > Taş

In [264]:
import random
secenekler =["taş","kağıt","makas"]
bilgsecimi=secenekler[random.randint(0,2)]
kullanicisecimi=input("taş/kağıt/makas seçin: ")
print("bilgisayar: ",bilgsecimi)
print("kullanıcı: ",kullanicisecimi)
if bilgsecimi =="taş" :
    if kullanicisecimi=="taş":
        print("berabere")
    if kullanicisecimi=="makas":
        print("bilg kazandı")
    if kullanicisecimi=="kağıt":
        print("kullanıcı kazandı")
        
if bilgsecimi =="makas" :
    if kullanicisecimi=="taş":
        print("kullanıcı kazandı")
    if kullanicisecimi=="makas":
        print("berabere")
    if kullanicisecimi=="kağıt":
        print("bilg kazandı")
        
if bilgsecimi =="kağıt" :
    if kullanicisecimi=="taş":
        print("bilg kazandı")
    if kullanicisecimi=="makas":
        print("kullanıcı kazandı")
    if kullanicisecimi=="kağıt":
        print("berabere")

taş/kağıt/makas seçin: makas
bilgisayar:  taş
kullanıcı:  makas
bilg kazandı


Zar atma oyunu
Kullanıcı "zar at" seçimi yaptığında bilgisayardan 2 tane sayı üretilir ve karşılaştırılır


In [269]:
import random
bilgisayarzar1 = random.randint(1,6)
bilgisayarzar2 = random.randint(1,6)
kullanicizar1 = random.randint(1,6)
kullanicizar2 = random.randint(1,6)
print("bilgisayar: ",bilgisayarzar1,"-",bilgisayarzar2)
print("kullanıcı: ",kullanicizar1,"-",kullanicizar2)
if bilgisayarzar1+bilgisayarzar2>kullanicizar1+kullanicizar2:
    print("bilgisayar kazandı")
else:
    print("kazandınız!")

bilgisayar:  5 - 6
kullanıcı:  3 - 6
bilgisayar kazandı


Girilen iki sözcüğün anagram olup olmadığını bulma
Anagram, aynı harflerden oluşan farklı sözcükler

In [270]:
sozcuk1 = sorted(input("sözcük gir: "))
sozcuk2 = sorted(input("sözcük gir: "))
if sozcuk1 == sozcuk2:
    print("anagram")
else:
     print("anagram değil")

sözcük gir: naz
sözcük gir: zan
anagram


Listede yinelenen öğeleri kaldırmak için bir Python programı yazın (listede her eleman eşşiz olacak)

In [272]:
liste = [10, 20, 30, 20, 10, 50, 60, 40, 80, 50, 40]
essizListe = []
for x in liste:
    if x not in essizListe:
        essizListe.append(x)
print(essizListe) 

[10, 20, 30, 50, 60, 40, 80]


In [ ]:
Bir sayıya kadar olan tüm asal sayıları bulan programı yazın
Liste’den tek tek asallık kontrolü yapmak?
Eratosthenes (Eratosten) kalburu olarak adlandırılan belirli bir tam sayıya kadar yer alan asal sayıların bulunması için kullanılan yöntemi kullanın
Önce bir listeye 2'den başlayarak, istediğiniz en büyük tam sayıya kadar olan tüm tam sayıları yazın. Bu listenin adı A olsun
Bir diğer listeye A'daki ilk asal sayı olan 2'den başlayarak bulduğunuz asal sayıları yazacaksınız. Bu listenin adı B olsun
A'dan 2'yi ve 2'nin tüm katlarını silin
A'da kalan ilk tek sayı asaldır. Bu sayıyı B'ye ekleyin
Bu sayıyı ve tüm katlarını A'dan silin. Daha küçük katları zaten silindiğinden, silme safhası bu sayının karesinden başlayabilir.
A listesinde herhangi bir sayı kalmayıncaya kadar 4. ve 5. adımları tekrarlayın

In [274]:
def eratosten(n):
    sayilar = []
    asalSayilar = []
    for i in range(2,n):
        sayilar.append(i)
    asalSayilar.append(2)
    for sayi in sayilar:
        if sayi%2==0:
            sayilar.remove(sayi)
    while(len(sayilar)>0):
        ilkSayi = sayilar[0]
        asalSayilar.append(ilkSayi)
        for sayi in sayilar:
            if sayi%ilkSayi==0:
                sayilar.remove(sayi)
    return asalSayilar
print(eratosten(100))

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]


In [ ]:
Fibonacci Dizisini verilen terim sayısına kadar yazdıran bir Python programı yazın
Fibonacci dizisi, her sayının kendinden önceki ile toplanması sonucu oluşan bir sayı dizisidir. 
1, 2, 3, 5, 8, 13, 21, 34, 55, 89... şeklinde devam eden bir dizi
Fibonacci Dizisini verilen terim sayısına kadar yazdıran bir Python programı yazın
Fibonacci dizisi, her sayının kendinden önceki ile toplanması sonucu oluşan bir sayı dizisidir. 
1, 2, 3, 5, 8, 13, 21, 34, 55, 89... şeklinde devam eden bir dizi
Bu durumda genel olarak n'inci Fibonacci sayısı F(n) şu şekilde ifade edilir:


In [275]:
def fibonacci(n):
    if(n <= 1):
        return n
    else:
        return(fibonacci(n-1) + fibonacci(n-2))
n = int(input("Terim sayısı girin:"))
for i in range(n):
    print(fibonacci(i),end="-")

Terim sayısı girin:5
0-1-1-2-3-